llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.cpp

```diff
@@ -337,9 +349,11 @@ unsigned GCNTTIImpl::getMinVectorRegisterBitWidth() const {
 unsigned GCNTTIImpl::getMaximumVF(unsigned ElemWidth, unsigned Opcode) const {
   if (Opcode == Instruction::Load || Opcode == Instruction::Store)
     return 32 * 4 / ElemWidth;
-  return (ElemWidth == 16 && ST->has16BitInsts()) ? 2
-       : (ElemWidth == 32 && ST->hasPackedFP32Ops()) ? 2
-       : 1;
+
+  return (ElemWidth == 8)                              ? 4
```

Comment:
Can you get away with just this part?

Reply:
Not really, no -- 

SLP will only attempt to vectorize candidates which have a vectorization factor greater than the number of parts (https://github.com/llvm/llvm-project/blob/ffca4ef5b1a8eff6097454df4b0f212e2393e41e/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp#L16246). So, while we could theoretically increase the maximum VF to 8 or 16, this would only vectorize shuffles of size 8 or 16, and miss vectorization on the most important case of v4i8.

Reply:
Did you have a patch where you tried to make v4i8 legal somewhere?

Reply:
Only a local branch -- when I was working on it (a long while ago) I couldn't find a way around making i8s legal, but I could always take a look. This whole chain of work is a workaround for v4i8 not being legal, and there are still other areas that need work arounds -- mfmas using i32s in place of v4i8 and the weird isel patterns that causes come to mind.

Even if we were to reinvestigate that though, I think we should implement the workaround and delete later.